File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h2x/system/calibrate_mech_zero_pos.cpp
// calibrate_mech_zero_pos.cpp

#include <iostream>
#include "h2x/h2x_system.h"
#include "common/h2x_def.h"

using namespace humanoid;
using namespace humanoid::h2x;

int main() {
  std::cout << "=== 关节机械零位标定示例 ===" << std::endl;
  H2xSystem system;
  const int joint_index = JointIndexValue::L_HIP_PITCH;

  int ret = system.calibrateZero(joint_index);
  std::cout << "标定关节 " << static_cast<uint32_t>(joint_index)
            << (ret == 0 ? " 成功" : " 失败") << std::endl;

  ret = system.saveJointParameters(joint_index);
  std::cout << "保存关节 " << static_cast<uint32_t>(joint_index)
            << (ret == 0 ? " 成功" : " 失败") << std::endl;

  return 0;
}

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h2x/system/calibrate_mech_zero_pos.py
import humanoid_sdk_py
import humanoid_sdk_py.h2x as h2x
import time

h2x_system = h2x.H2xSystem()
time.sleep(1)

print("=== 关节机械零位标定示例 ===")
joint_index = h2x.L_HIP_PITCH

ret = h2x_system.calibrateZero(joint_index)
print("标定关节 ", joint_index, ("成功" if ret == 0 else " 失败"))

ret = h2x_system.saveJointParameters(joint_index)
print("保存关节 ", joint_index, ("成功" if ret == 0 else " 失败"))
